### PR TITLE
BigInt: Add tests for BigInt conversions in TypedArray prototype/set and TypedArray internal [[set]]

### PR DESCRIPTION
**Pull request by spectranaut**

### CODE REVIEW

File: test/built-ins/TypedArrays/internals/Set/BigInt/number-tobigint.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2018 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid:  sec-assignment-operators-runtime-semantics-evaluation
+description: >
+  Return abrupt on Number
+info:
```
Comment (leobalter) [inline test/built-ins/TypedArrays/internals/Set/BigInt/number-tobigint.js]: the missing pipe `|` is breaking the yaml here.

File: test/built-ins/TypedArray/prototype/set/BigInt/number-tobigint.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2018 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.set-array-offset
+description: >
+  Return abrupt on Number
+info:
```
Comment (leobalter) [inline test/built-ins/TypedArray/prototype/set/BigInt/number-tobigint.js]: same problem with a missing pipe

File: test/built-ins/TypedArray/prototype/set/BigInt/src-typedarray-big.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  Sets multiple values in this TypedArray, reading the values from the
+  typedArray argument object. The optional offset value indicates the first
+  element index in this TypedArray where values are written. If omitted, it
+  is assumed to be 0.
+  ...
+  23. If one of srcType and targetType contains the substring "Big" and the
+      other does not, throw a TypeError exception.
+  ...
+
+includes: [testBigIntTypedArray.jsx]
```
Comment (leobalter) [inline test/built-ins/TypedArray/prototype/set/BigInt/src-typedarray-big.js]: .js :)

File: test/built-ins/TypedArray/prototype/set/BigInt/bigint-tobigint64.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  -18446744073709551618n, // -(2n ** 64n) - 2n
+];
+
+var typedArray = new BigInt64Array(vals.length);
+typedArray.set(vals);
+
+assert.sameValue(typedArray[0], 2n,
+                 "ToBigInt64(2n ** 64n + 2n) => 2n");
+
+assert.sameValue(typedArray[1], -9223372036854775806, // 2n - 2n ** 63n
```
Comment (spectranaut) [inline test/built-ins/TypedArray/prototype/set/BigInt/bigint-tobigint64.js]: the inconsistencies are because I forgot the "n" here XD

**Pull request merged**